File: src/main/java/com/mohe/spring/dto/ActivityDto.java
package com.mohe.spring.dto;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

public class ActivityDto {

    // Recent Places
    public record RecentPlacesResponse(
        List<RecentPlaceData> recentPlaces
    ) {}

    public record RecentPlaceData(
        String id,
        String title,
        String location,
        String image,
        BigDecimal rating,
        OffsetDateTime viewedAt
    ) {}

    // My Places
    public record MyPlacesResponse(
        List<MyPlaceData> myPlaces
    ) {}

    public record MyPlaceData(
        String id,
        String title,
        String location,
        String image,
        String status,
        OffsetDateTime createdAt
    ) {}
}

File: src/main/java/com/mohe/spring/dto/ActivityDto.kt
package com.mohe.spring.dto

import java.math.BigDecimal
import java.time.OffsetDateTime

// Recent Places
data class RecentPlacesResponse(
    val recentPlaces: List<RecentPlaceData>
)

data class RecentPlaceData(
    val id: String,
    val title: String,
    val location: String?,
    val image: String?,
    val rating: BigDecimal,
    val viewedAt: OffsetDateTime
)

// My Places
data class MyPlacesResponse(
    val myPlaces: List<MyPlaceData>
)

data class MyPlaceData(
    val id: String,
    val title: String,
    val location: String?,
    val image: String?,
    val status: String,
    val createdAt: OffsetDateTime
)